### PR TITLE
adds support for ALTER TABLE

### PR DESCRIPTION
**Pull request by brunocalza**
# Summary

bumps the parser to add support for ALTER TABLE

# Context

See https://github.com/tablelandnetwork/go-sqlparser/pull/55

# Implementation overview

It's a straightforward add:
- bumps the parser
- adjusts the validator parsing logic to consider ALTER TABLE operation in write statement
- adjust the runsql executor to only execute ALTER TABLE if you are the owner
- adds tests on 3 differents layers: parsing layer, executor layer and e2e

### CODE REVIEW

File: pkg/eventprocessor/impl/executor/impl/txnscope_runsql.go [hunk 1]
```diff
@@ -220,7 +220,17 @@ func (ts *txnScope) executeWriteStmt(
 	addr common.Address,
 	policy tableland.Policy,
 	beforeRowCount int,
+	isOwner bool,
 ) error {
+	if ws.Operation() == tableland.OpAlter {
```
Comment (brunocalza) [inline pkg/eventprocessor/impl/executor/impl/txnscope_runsql.go:225]: only the owner can execute the ALTER TABLE operation. that's the only ACL rule we have right now. ALTER TABLE is not part of the GRANT/REVOKE permissions and neither part of the policy

File: pkg/eventprocessor/impl/executor/impl/txnscope_runsql_test.go [hunk 2]
```diff
@@ -423,6 +423,48 @@ func TestRunSQL_WriteQueriesWithPolicies(t *testing.T) {
 	})
 }
 
+func TestRunSQL_AlterTable(t *testing.T) {
```
Comment (brunocalza) [inline pkg/eventprocessor/impl/executor/impl/txnscope_runsql_test.go:426]: this test checks when a non owner tries to execute an ALTER TABLE

File: pkg/parsing/impl/validator_test.go [hunk 3]
```diff
@@ -218,6 +218,34 @@ func TestWriteQuery(t *testing.T) {
 			expErrType: nil,
 		},
 
+		// ALTER TABLE
```
Comment (brunocalza) [inline pkg/parsing/impl/validator_test.go:221]: adds tests at the parsing level

File: pkg/parsing/impl/validator.go [hunk 4]
```diff
@@ -166,6 +166,9 @@ func (pp *QueryValidator) ValidateMutatingQuery(
 			if _, ok := s.(*sqlparser.Delete); ok {
 				mutatingStmt.operation = tableland.OpDelete
 			}
+			if _, ok := s.(*sqlparser.AlterTable); ok {
```
Comment (brunocalza) [inline pkg/parsing/impl/validator.go:169]: marks the operation of the statement with`tableland.OpAlter`

**Review by avichalp — APPROVED** [review]
LGTM

**Pull request merged**